src/mo_frontend/typing.ml

```diff
@@ -3809,6 +3912,7 @@ and infer_dec_valdecs env dec : Scope.t =
       typ_env = T.Env.singleton id.it c;
       con_env = T.ConSet.singleton c;
     }
+  | MixinD (_, _) -> Scope.empty
```

Comment:
This seems slightly dubious, but I guess a top-level mixin is still checked somewhere or nothing would work.

Should this be an assert if !env.in_prog?